nntrainer/models/neuralnet.cpp

```diff
@@ -396,21 +403,24 @@ void NeuralNetwork::load(const std::string &file_path,
       /// read. so, after this line, additional read shouldn't be called
       model_file.seekg(bin_file_pos);
 
-      if (istrequal(opt->getType(), "adam")) {
-        char opt_type[4];
-        model_file.read(opt_type, 4);
-        if (istrequal(opt_type, "adam")) {
-          for (auto iter = model_graph.cbegin(); iter != model_graph.cend();
-               iter++) {
-            (*iter)->read(model_file, true);
-          }
-        }
-      }
-
       checkedRead(model_file, (char *)&epoch_idx, sizeof(epoch_idx),
                   "[NeuralNetwork::readModel] failed to read epoch_idx");
       checkedRead(model_file, (char *)&iter, sizeof(iter),
                   "[NeuralNetwork::readModel] failed to read iteration");
+
+      opt->read(model_file);
```

Comment:
this is used to be string of "adam", now it's single value, this is non backward compatible change which needs attention.